airflow/providers/common/sql/operators/sql.py

```diff
@@ -273,38 +303,38 @@ def __init__(
 
         self.table = table
         self.checks = checks
+        self.partition_clause = partition_clause
         # OpenLineage needs a valid SQL query with the input/output table(s) to parse
         self.sql = f"SELECT * FROM {self.table};"
 
     def execute(self, context=None):
         hook = self.get_db_hook()
-
-        check_names = [*self.checks]
-        check_mins_sql = ",".join(
-            self.sql_min_template.replace("check_name", check_name) for check_name in check_names
-        )
-        checks_sql = ",".join(
+        checks_sql = " UNION ALL ".join(
             [
-                self.sql_check_template.replace("check_statement", value["check_statement"]).replace(
-                    "check_name", check_name
-                )
+                self.sql_check_template.replace("check_statement", value["check_statement"])
+                .replace("_check_name", check_name)
+                .replace("table", self.table)
                 for check_name, value in self.checks.items()
             ]
         )
+        partition_clause_statement = f"WHERE {self.partition_clause}" if self.partition_clause else ""
+        self.sql = f"SELECT check_name, check_result FROM ({checks_sql}) "
+        f"AS check_table {partition_clause_statement};"
 
-        self.sql = f"SELECT {check_mins_sql} FROM (SELECT {checks_sql} FROM {self.table});"
-        records = hook.get_first(self.sql)
+        records = hook.get_pandas_df(self.sql)
```

Comment:
Why did we change from getting records to getting this as a pandas dataframe? This now places a _hard_ requirement on using pandas for this operator, where as previously pandas was almost entirely optional.

Reply:
So, this was changed because with `hook.get_first`, there was an issue with how the SQL was being written that caused *only* fully aggregated checks to be returned, unless the syntax of the SQL query was changed, but that would require either a `fetch_all` or `get_pandas_df` call as the new SQL needs to returned multiple lines. It seemed much easier and possibly more efficient to use pandas here, but if a `fetch_all` seems more reasonable this can be changed. Happy to explain more about the specific issue if curious.

Reply:
I'd be curious to hear more about it :D

Reply:
To expand a bit more, a check like `col_a + col_b >= col_c` would not work when there were multiple checks in the operator as the previous `SELECT` statement would then fail and require a `GROUP BY` clause iirc. So the check would either have to be in its own operator, or be amended like so: `SUM(col_a) + SUM(col_b) >= SUM(col_c)` which isn't quite the same check. So the query needed to be updated, and the one that I wound up using returns multiple rows. So `get_first` is no longer useful, and in the moment of writing it seemed that handling things with a pandas dataframe might be easier in the long term, if more complicated uses of the pulled in data were implemented. But as of now I see how it's unneeded.

Reply:
Thanks.... Is the operator easy to understand by the users? I am afraid this is something we need good and clear howto and examples for it because people will have hard time using it and rais too many questions.

Reply:
Ideally most users won't need to learn about why `.fetch_all` is being used instead of `.get_first` 🙃 . I'm working with some users of the operator right now and seeing what's complicated to make sure the docs are robust. I also have a working example DAGs showing how to use the operator (with several more planned) [here](https://registry.astronomer.io/providers/common-sql/modules/sqltablecheckoperator/#example-dags).